### PR TITLE
Add interpreter fast paths for hot-path operations

### PR DESCRIPTION
**Pull request by pmatos**
## Summary

- Add targeted interpreter fast paths to reduce overhead for computation-heavy workloads (refs #54)
- Short-circuit the common `Number` case across arithmetic, property access, variable lookup, and assignment operations
- Consolidate identifier resolution from 3 separate scope chain walks into 1 combined pass
- test262: 99,020/99,020 (100.00%), 0 regressions, 10 new passes

### Optimizations

| Fast path | File | Description |
|-----------|------|-------------|
| `eval_binary` Number+Number | `eval.rs` | Skip ToPrimitive/ToNumeric/BigInt checks for all arithmetic, bitwise, comparison ops |
| `to_property_key` integer | `literals.rs` | Use `u32::to_string()` instead of ryu_js for non-negative integer indices |
| `eval_member` typed array/array read | `access.rs` | Bypass `to_property_key` + `get_object_property` for numeric indices |
| `eval_assign` typed array write | `eval.rs` | Skip `to_property_key` for numeric typed array writes |
| `eval_assign` local binding | `eval.rs` | Simple + compound assignment skip `resolve_identifier_ref` for local `var`/`let` |
| `eval_update` local Number | `eval.rs` | `i++`/`i--` directly increment without full identifier resolution |
| `resolve_identifier` single-pass | `eval.rs` | 1 scope walk instead of 3 (with-check + global-getter + env.get) |
| `get_object_property` single-borrow | `modules.rs` | Classify object in 1 borrow; ordinary objects skip proxy/module checks |

### Microbenchmark results

| Benchmark | Before | After | Improvement |
|-----------|--------|-------|-------------|
| Empty 1M loop | 1,727ms | ~1,250ms | 28% |
| 1M arithmetic+bitwise (global) | 3,275ms | ~2,400ms | 27% |
| 1M arithmetic (function-scoped) | ~2,000ms | 1,372ms | 31% |
| 2.6M typed array reads | 12,803ms | ~8,100ms | 37% |

### Note on mandreel

The mandreel benchmark still times out at 120s. These interpreter-level fast paths provide meaningful speedups but the remaining gap requires deeper architectural changes tracked in separate issues.

## Test plan

- [x] `./scripts/lint.sh` passes (clippy clean)
- [x] Full test262 suite: 99,020/99,020 (100.00%), 0 regressions
- [x] Smoke tests for edge cases: `const` assignment rejection, `with` statement, TDZ, `Object.defineProperty` on array indices
- [x] Microbenchmarks show improvement across arithmetic, typed array, and loop patterns

### CODE REVIEW

File: src/interpreter/eval.rs [hunk 1]
```diff
@@ -2649,6 +2746,27 @@ impl Interpreter {
...
+                                    b.kind == BindingKind::Var || b.kind == BindingKind::Let
+                                })
+                        };
+                        if has_mutable_local {
+                            let rval = match self.eval_expr(right, env) {
+                                Completion::Normal(v) => v,
+                                other => return other,
+                            };
+                            let _ = env.borrow_mut().set(name, rval.clone());
+                            return Completion::Normal(rval);
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/interpreter/eval.rs]: Fixed in `af18f6c` — now propagates errors from `env.set()` via `match Ok/Err` instead of `let _ =`.

File: src/interpreter/eval.rs [hunk 2]
```diff
@@ -2538,6 +2609,32 @@ impl Interpreter {
...
+                            };
+                            let obj_ref = obj.borrow();
+                            let ta = obj_ref.typed_array_info.as_ref().unwrap();
+                            if is_valid_integer_index(ta, *index) {
+                                let ta_clone = ta.clone();
+                                drop(obj_ref);
+                                typed_array_set_index(&ta_clone, *index as usize, &num_val);
+                            }
+                            return Ok(());
+                        }
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/interpreter/eval.rs:2636]: Fixed in `af18f6c` — early return now only happens when `is_valid_integer_index` succeeds. Invalid indices (e.g. `-0`, NaN, fractional) fall through to `to_property_key` for correct canonicalization.

File: src/interpreter/eval.rs [hunk 3]
```diff
@@ -2230,6 +2274,33 @@ impl Interpreter {
...
+                {
+                    let old_num = *n;
+                    let new_num = match op {
+                        UpdateOp::Increment => old_num + 1.0,
+                        UpdateOp::Decrement => old_num - 1.0,
+                    };
+                    let new_val = JsValue::Number(new_num);
+                    drop(env_borrow);
+                    let _ = env.borrow_mut().set(name, new_val);
+                    return Completion::Normal(JsValue::Number(if prefix {
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/interpreter/eval.rs]: Fixed in `aec4592` — now propagates errors from `env.set()` via `if let Err(e)` instead of `let _ =`.
